src/commands/optimize.ts

```diff
@@ -40,6 +40,7 @@ export default class Optimize extends Command {
     optimization: Flags.string({char: 'p', default: Object.values(Optimizations), options: Object.values(Optimizations), multiple: true, description: 'select the type of optimizations that you want to apply.'}),
     output: Flags.string({char: 'o', default: Outputs.TERMINAL, options: Object.values(Outputs), description: 'select where you want the output.'}),
     'no-tty': Flags.boolean({ description: 'do not use an interactive terminal', default: false }),
+    disableOptimizationForSchema: Flags.boolean({ description: 'schema should be ignored from the optimization prcoess', default: false }),
```

Comment:
Yes, even I was thinking that. I just used this since it was suggested here https://github.com/asyncapi/cli/issues/1432#issuecomment-2120377125

Reply:
makes sense. In the original issue, I suggested `--ignore` but `--omit` also works.

Reply:
Just for confirmation @Amzani, when you mention other things that can be omitted by changing the flag format, do you mean this, like `reuseComponents`, `removeComponents` etc can also be enabled or disabled by default?
```
{
  output: 'YAML',
  rules: {
    reuseComponents: true,
    removeComponents: true,
    moveAllToComponents: true,
    moveDuplicatesToComponents: false,
  },
  disableOptimizationFor: {
    schema: false,
  },
}
```

Reply:
@KhudaDad414 is the best person to answer this one.

Reply:
@helios2003 currently optimizer can only disable the schemas. but if in the future we want to disable optimizations for other components like `messages` or `channels` we can have the option.

`reuseComponents`, `removeComponents` are rules and this flag that you are supporting doesn't have anything to do with them.

Hope it clears the confusion.

Reply:
@KhudaDad414, Should these changes be made directly in the optimizer repository, or is it sufficient to implement them locally in the CLI for now since I noticed that the current optimizer repository does not seem to support disabling optimization for components other than the schema https://github.com/asyncapi/optimizer/blob/b363a580288abc5dc522814ef9d719b9dae929d9/src/types.ts#L40.

Reply:
@helios2003
Yes,

- Change
https://github.com/asyncapi/optimizer/blob/b363a580288abc5dc522814ef9d719b9dae929d9/src/Optimizer.ts#L120
as per
https://github.com/asyncapi/cli/issues/1432#issuecomment-2120377125
```diff
- disableOptimizationFor: {
-   schema: false,
- },
+ disableOptimizationForSchema: false
```

- Delete interface `DisableOptimizationFor`

- In the interface `Options`
```diff
- disableOptimizationFor?: DisableOptimizationFor
+ disableOptimizationForSchema?: boolean
```

to make the name of the switch `disableOptimizationForSchema` consistent across standalone and CLI versions.

@KhudaDad414
Am I right?

Reply:
@helios2003 @aeworxet tbh I don't have a strong opinion here. one one side 

```
disableOptimizationFor: {
  schema: false,
}
```
can be used to add more components to ignore in the future. things like:

```
disableOptimizationFor: {
  schema: false,
  message: true
}
```
the question is whether it's a valid case? will someone want this feature? I don't know.

and on the other hand changing the name to `disableOptimizationForSchema` would make a more seemless way of integration between CLI and optimizer.

Reply:
@KhudaDad414

Would `--disableOptimizationFor='{"schema":"false","message":"true"}'` be too much?

Reply:
> @KhudaDad414
> 
> Would `--disableOptimizationFor='{"schema":"false","message":"true"}'` be too much?

Instead of doing it like this, can't we do it like ``--ignore=schema, message`` etc.  Whatever we write after the ``=`` it what we want to disable the optimization for.

Reply:
Oh, okay, I see, but even if that is the case, passing the flags as JSON in the command line isn't usually done from what I have seen, so having it otherwise might help.

Reply:
During all the time I've been working with `Optimizer`, I didn't see a need to disable optimization for anything else except `schema`, and that was only because schemas were disrupting the view of optimized AsyncAPI Document. So I think if at least one user requests anything else, it would be possible to think about changes to the logic, and for now it can be simply the introduction of one consistent across versions switch `disableOptimizationForSchema` and that's it.

@KhudaDad414 wdyt?

Reply:
> @helios2003 @aeworxet tbh I don't have a strong opinion here. one one side
> 
> ```
> disableOptimizationFor: {
>   schema: false,
> }
> ```
> 
> can be used to add more components to ignore in the future. things like:
> 
> ```
> disableOptimizationFor: {
>   schema: false,
>   message: true
> }
> ```
> 
> the question is whether it's a valid case? will someone want this feature? I don't know.
> 
> and on the other hand changing the name to `disableOptimizationForSchema` would make a more seemless way of integration between CLI and optimizer.

I am bit confused. So do I need to add additional components to the flags and change it in both the repositories or just do it for `schema`?

Reply:
@helios2003 since optimizer currently only supports `schemas` I think you can only support schemas here and call it a day.